### PR TITLE
Enable Mypy Check in torch/_inductor/triton_heuristics.py

### PR DESCRIPTION
**Pull request by fffrog**
Fixes #105230

```shell
$ lintrunner init && lintrunner -a torch/_inductor/triton_heuristics.py
...
ok No lint issues.
Successfully applied all patches.
```

```shell
$ mypy torch/_inductor/triton_heuristics.py
Success: no issues found in 1 source file
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

**Comment by colesbury** [comment]
@eellison, I'm requesting reviews from you for PRs that reference #105230. Let me know if there are other people I should instead request reviews from.

**Comment by eellison** [comment]
@colesbury sounds good, you can just assign it.

File: torch/_inductor/triton_heuristics.py [hunk 1]
```diff
@@ -396,7 +397,7 @@ def run(self, *args, grid, stream):
 
         if launcher.config.pre_hook is not None:
             launcher.config.pre_hook(
-                {**zip(self.arg_names, args), **launcher.config.kwargs}
+                {*zip(self.arg_names, args), *launcher.config.kwargs}
```
Comment (eellison) [inline torch/_inductor/triton_heuristics.py:400]: is this not a change in sematnics ?

Comment (FFFrog) [reply torch/_inductor/triton_heuristics.py]: Sorry, I submitted a new commit to fix it.

File: torch/_inductor/triton_heuristics.py [hunk 2]
```diff
@@ -396,7 +397,7 @@ def run(self, *args, grid, stream):
 
         if launcher.config.pre_hook is not None:
             launcher.config.pre_hook(
-                {**zip(self.arg_names, args), **launcher.config.kwargs}
+                {**dict(zip(self.arg_names, args)), **launcher.config.kwargs}
```
Comment (ezyang) [inline torch/_inductor/triton_heuristics.py:400]: this is weird and possibly worse than before?

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**